homeassistant/components/daikin.py

```diff
@@ -0,0 +1,294 @@
+
+SENSOR_TYPES = {
+    ATTR_INSIDE_TEMPERATURE: {
+        CONF_NAME: 'Inside Temperature',
+        CONF_ICON: 'mdi:thermometer'
+    },
+    ATTR_OUTSIDE_TEMPERATURE: {
+        CONF_NAME: 'Outside Temperature',
+        CONF_ICON: 'mdi:thermometer'
+    }
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_HOSTS, default=[]): vol.Schema([cv.string]),
+        vol.Optional(
+            CONF_MONITORED_CONDITIONS,
+            default=list(SENSOR_TYPES.keys())
+        ): vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+HA_STATE_TO_DAIKIN = {
+    STATE_FAN_ONLY: 'fan',
+    STATE_DRY: 'dry',
+    STATE_COOL: 'cool',
+    STATE_HEAT: 'hot',
+    STATE_AUTO: 'auto',
+    STATE_OFF: 'off',
+}
+
+KNOWN_DEVICES = []
+
+
+def setup(hass, config):
+    """Establish connection with Daikin."""
+    def discovery_dispatch(service, discovery_info):
+        """Dispatcher for Daikin discovery events."""
+        host = discovery_info.get('ip')
+
+        if manual_device_setup(hass, host) is None:
+            return
+
+        for component in COMPONENT_TYPES:
+            load_platform(hass, component, DOMAIN, discovery_info,
+                          config)
+
+    discovery.listen(hass, SERVICE_DAIKIN, discovery_dispatch)
+
+    devices = []
+
+    # Add static devices from the config file.
+    devices.extend(
+        (host, None)
+        for host in config.get(DOMAIN, {}).get(CONF_HOSTS, [])
+    )
+
+    for host in devices:
+        if manual_device_setup(hass, host) is not None:
+            discovery_info = {
+                'ip': host,
+                CONF_MONITORED_CONDITIONS:
+                    config.get(DOMAIN, {}).get(CONF_MONITORED_CONDITIONS)
+            }
+            load_platform(hass, 'sensor', DOMAIN, discovery_info, config)
+
+    return True
+
+
+def manual_device_setup(hass, host, name=None):
+    """Create a Daikin instance only once."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    if hass.data[DOMAIN].get(host) is None:
+        from pydaikin import appliance
+
+        try:
+            device = appliance.Appliance(host)
+        except timeout:
+            _LOGGER.error("Connection to Daikin could not be established")
+            return False
+
+        if name is None:
+            name = device.values['name']
+
+        hass.data[DOMAIN][host] = DaikinEntity(device, name)
+
+    return hass.data[DOMAIN][host]
+
+
+class DaikinEntity(object):
+    """Keep the Daikin instance in one place and centralize the update."""
+
+    def __init__(self, device, name):
+        """Initialize the Daikin Handle."""
+        from pydaikin import appliance
+
+        self.device = device
+        self.name = name
+        self.ip_address = device.ip
+
+        self.mutex = Lock()
+        self._scan_interval = DEFAULT_SCAN_INTERVAL
+        self._last_update = time.time()
+
+        self._operation_list = list(
```

Comment:
This belongs into the climate platform, including `_fan_list` and `_swing_list`.

Reply:
Done